api/v1beta1/foundationdbcluster_types.go

```diff
@@ -1361,11 +1364,17 @@ func (cluster *FoundationDBCluster) CheckReconciliation() (bool, error) {
 	cluster.Status.Generations = ClusterGenerationStatus{Reconciled: cluster.Status.Generations.Reconciled}
 
 	for _, processGroup := range cluster.Status.ProcessGroups {
-		if processGroup.Remove && !processGroup.Excluded {
-			cluster.Status.Generations.NeedsShrink = cluster.ObjectMeta.Generation
-			reconciled = false
-		} else if processGroup.Remove {
-			cluster.Status.Generations.HasPendingRemoval = cluster.ObjectMeta.Generation
+		if processGroup.Remove {
+			terminating := false
+			for _, condition := range processGroup.ProcessGroupConditions {
+				terminating = terminating || condition.ProcessGroupConditionType == ResourcesTerminating
+			}
+			if processGroup.Excluded && terminating {
```

Comment:
```suggestion
			if processGroup.Excluded || terminating {
```

That should rather be an or. We could have an instance that is in the terminating state but not excluded (e.g. removal without exclusion). I'm also not sure if we really care about the excluded state, I would argue that we only care about the terminating state because if the process group is in a terminating state that means that the process group was either successfully excluded or we don't want to exclude it.

Reply:
And in the current implementation we never set `processGroup.Excluded` to true and update the field. I created an issue for that.

Reply:
If a process group is excluded but not in a terminating state, we don't want to set `HasPendingRemoval` yet.

Reply:
As it stands today, we could probably just rely on `terminating` here. I'm not sure what behavior we want if a process group is marked for removal, but isn't excluded yet, and the pod gets deleted for some other reason.

As an alternative here, we could bring in the `ExclusionSkipped` flag.

Reply:
I think for now just using `terminating` should be good. We could open a follow up issue and once we populate the exclude flag again (https://github.com/FoundationDB/fdb-kubernetes-operator/pull/800) we can change that to:

`(processGroup.Excluded || processGroup. ExclusionSkipped) && terminating `